mlir/include/mlir/Conversion/Passes.td

```diff
@@ -733,6 +733,24 @@ def ConvertMathToLLVMPass : Pass<"convert-math-to-llvm"> {
   ];
 }
 
+//===----------------------------------------------------------------------===//
+// MathToLibm
+//===----------------------------------------------------------------------===//
+
+def ConvertMathToROCDL : Pass<"convert-math-to-rocdl", "ModuleOp"> {
```

Comment:
This pass doesn't need to operate on a `ModuleOp` specifically - you probably want this to work on anything that has a symbol table

Reply:
It needs to have a DataLayoutOpInterface as well. Should this be checked dynamically, so that the pass can be used on any operation but should only be triggered on ops with a symbol table and data layout interface?

Reply:
Hm, yeah, good point

(I'm basically trying to make sure this works on `gpu.module` and `builtin.module`)

Reply:
Yes, I was thinking about making it work for both as well, but opted to use the "add patterns" function in the GPUToROCDL conversion since the options and converter are more complicated, and I didn't want to put that in this simpler pass, so it seemed okay for it to operate on the ModuleOp. I can try to see if I can make it work with both if you need it.

Reply:
... Good point, this pass is for testing anyway, let's keep it at `ModuleOp`